### DISCUSSION TITLE
Only generate 3D elements

### DISCUSSION THREAD

**Discussion by dthillaithevan** [nschloe/pygmsh#524] [Q&A] [answered]
Hi, 

Apologies if this is a trivial question - is there anyway to ensure 3D meshes contain only 3D tetra elements (no triangles, lines or points)? 

Is there a command I am missing, e.g. how should I modify the code below to ensure only tetrahedrals are generated? 

```
import pygmsh
with pygmsh.occ.Geometry() as geom:
    geom.characteristic_length_max = 0.05
    geom.add_cylinder([0.0, 0.0, 0.0], [0.0, 0.1, 0.0], 0.4)
    mesh = geom.generate_mesh()

# Output mesh contains the following: 
#   Number of points: 732
#   Number of cells:
#   line: 104
#   triangle: 1224
#   tetra: 2405 <---
#   vertex: 2
```

Thanks!

**Comment by dthillaithevan** [comment, marked as answer]
I think I found a solution, not sure if this is the correct way but it seems to work so far. 

```
elems = mesh.cells[2].data
coords = mesh.points

cells = [("tetra", elems)]
meshio.write_points_cells('test.vtk',coords,cells)
```

**Reply by nschloe** [reply]
Yeah, that would work. I'd check beforehand that `mesh.cells[2]` are the only tetra elements, but perhaps you already did that.

**Reply by dthillaithevan** [reply]
Yeah I did check that, thank you!
